lib/logstash/inputs/snmptrap.rb

```diff
@@ -47,26 +47,44 @@ def register
     end
   end # def register
 
-  public
   def run(output_queue)
     begin
       # snmp trap server
       snmptrap_listener(output_queue)
     rescue => e
       @logger.warn("SNMP Trap listener died", :exception => e, :backtrace => e.backtrace)
-      sleep(5)
-      retry
+      stoppable_call(20) { sleep(0.25) }
```

Comment:
but I do not think we need to call wakeup here, breaking down a sleep(N) into a N * sleep(1) and checking at each iteration should be good enough. thoughts?